### ISSUE TITLE
Integration test for Users screen

### ISSUE THREAD

**Issue by teodosii** [grafana/oncall#1586]
Our frontend tests lack plenty of integration tests, it would be super nice to start adding a few in order to test the most used flows. This first integration test on Users should ideally cover

- permissions (Viewer/Editor/Admin when accessing links or tabs from View My Profile)
- actions on the UI do not throw exceptions

Ideally we should only mock the stores to have data in it, anything else should work in integration with other components part of the Users tree.

**Comment by joeyorlando** [comment]
@teodosii 👋 wdyt about re-describing the scope of this issue, to make the list of deliverables very clear? imo, it's a bit open-ended, so it's tough to know when this is task is complete.

Example deliverables could be something like:
- As an Admin or Editor, I should be able to do...
- As a Viewer, I should be not be able to..

**Comment by teodosii** [comment]
Good idea, I'll rephrase this task once I get back to it. Or if you wanna take this instead feel free to do it as well.

**Comment by maskin25** [comment]
Currently I added two tests:
- admin is allowed to edit other profiles
- editor is not allowed to edit other profiles

(see `grafana-plugin/integration-tests/users/usersActions.test.ts`)

**Comment by teodosii** [comment]
I'm thinking of these scenarios
- **Viewers** are not allowed to view other users
- **Viewers** cannot access restricted tabs in the view my profile
- **Editors** cannot edit other users' profiles
- **Editors** cannot view other users' data
- **Editors** cannot access restricted tabs in the view my profile
- **Admin** can do all the above
